### PR TITLE
Newsletter demo

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: apps/demo/src/scss/styles.scss [hunk 1]
```diff
@@ -5,8 +5,8 @@
 ************************************/
 
 // Imports
-@import 'daff-typography';
-@import 'daff-theme';
+@import 'daff-typography.scss';
```
Comment (Nolan-Arnold) [inline apps/demo/src/scss/styles.scss]: This change along with the changes in `theme.scss` and `utilities.scss` were to fix a bug on windows. Where the build could not differentiate between .css and .scss files.

File: libs/newsletter/src/effects/newsletter.effects.ts [hunk 2]
```diff
@@ -27,6 +30,14 @@ export class DaffNewsletterEffects<T extends DaffNewsletterSubmission, V>{
           return of(new DaffNewsletterFailedSubscribe('Failed to subscribe to newsletter'));
         })
       )
-    ))
-  )
+    )));
+
+  cancelSubmission$: Observable<Action> = createEffect(() => this.actions$.pipe(
```
Comment (damienwebdev) [inline libs/newsletter/src/effects/newsletter.effects.ts]: This needs to cancel the `trySubmission$` observable, it won't work as a standalone observable. This definitely needs test coverable a marble diagram. 

https://github.com/ReactiveX/rxjs/blob/master/docs_app/content/guide/testing/marble-testing.md
https://www.learnrxjs.io/operators/transformation/switchmap.html

File: apps/demo/src/app/newsletter/newsletter.component.html [hunk 3]
```diff
@@ -8,9 +8,12 @@ <h3 class="newsletter__title">
         Be the first to know about the latest products, exclusives, and offers!
       </div>
     </div>
-    <div class="newsletter__right">
-      <input daff-input type="text" class="newsletter__input" placeholder="Email" name="email" />
-      <button type="submit" daff-button color="secondary" class="newsletter__button">Sign Up</button>
+    <div>
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.html]: How do we indicate loading?

File: apps/demo/src/app/newsletter/newsletter.component.html [hunk 4]
```diff
@@ -8,9 +8,12 @@ <h3 class="newsletter__title">
...
+      <div class="newsletter__right" *ngIf="!(success$ | async)">
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.html]: How we do indicate that the user can cancel the request?

File: apps/demo/src/app/newsletter/newsletter.component.ts [hunk 5]
```diff
@@ -1,8 +1,28 @@
...
+  error$ = this.newsletterFacade.error$;
+  loading$ = this.newsletterFacade.loading$;
+
+  email: FormControl = new FormControl('');
+
+  constructor(public newsletterFacade: DaffNewsletterFacade) {
+  }
+  onNewsletterSubmit() {
+    this.newsletterFacade.dispatch(new DaffNewsletterSubscribe<DaffNewsletterSubmission>(this.email.value));
+    console.log('test2')
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.ts]: Remove.

File: apps/demo/src/app/newsletter/newsletter.component.ts [hunk 6]
```diff
@@ -1,8 +1,28 @@
...
+
+  success$ = this.newsletterFacade.success$;
+  error$ = this.newsletterFacade.error$;
+  loading$ = this.newsletterFacade.loading$;
+
+  email: FormControl = new FormControl('');
+
+  constructor(public newsletterFacade: DaffNewsletterFacade) {
+  }
+  onNewsletterSubmit() {
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.ts]: Needs test coverage.

File: apps/demo/src/scss/styles.scss (hunk 1 above)
Comment (damienwebdev) [reply apps/demo/src/scss/styles.scss]: I don't believe there is a `daff-typography.css` file?

Comment (damienwebdev) [reply apps/demo/src/scss/styles.scss]: @Nolan-Arnold and I resolved this offline and realised that he had a auto scss compiler installed.

File: apps/demo/src/app/newsletter/newsletter.module.ts [hunk 7]
```diff
@@ -4,19 +4,27 @@ import { CommonModule } from '@angular/common';
...
   ],
   declarations: [
-    NewsletterComponent
+    NewsletterComponent,
   ],
   exports: [
     NewsletterComponent
+  ],
+  providers: [
+    DaffNewsletterFacade
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.module.ts]: This shouldn't be provided here.

File: apps/demo/src/app/newsletter/newsletter.module.ts [hunk 8]
```diff
@@ -4,19 +4,27 @@ import { CommonModule } from '@angular/common';
...
     DaffInputModule,
     DaffButtonModule,
-    DaffContainerModule
+    DaffContainerModule,
+    DaffNewsletterModule,
+    ReactiveFormsModule
   ],
   declarations: [
-    NewsletterComponent
+    NewsletterComponent,
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.module.ts]: Remove the unnecessary comma.

File: apps/demo/src/app/newsletter/newsletter.component.ts [hunk 9]
```diff
@@ -1,8 +1,38 @@
...
   styleUrls: ['./newsletter.component.scss']
 })
-export class NewsletterComponent {}
+export class NewsletterComponent {
+
+  success$ = this.newsletterFacade.success$;
+  error$ = this.newsletterFacade.error$;
+  loading$ = this.newsletterFacade.loading$;
+
+  email: FormControl = new FormControl('');
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.ts]: This should validate whether or not the `email` is in fact an email.

File: apps/demo/src/app/newsletter/newsletter.component.html [hunk 10]
```diff
@@ -8,9 +8,23 @@ <h3 class="newsletter__title">
         Be the first to know about the latest products, exclusives, and offers!
       </div>
     </div>
-    <div class="newsletter__right">
-      <input daff-input type="text" class="newsletter__input" placeholder="Email" name="email" />
-      <button type="submit" daff-button color="secondary" class="newsletter__button">Sign Up</button>
+    <div>
+      <div class="newsletter__right" *ngIf="!(success$ | async) && !(error$ | async) && !(loading$ | async)">
+        <input daff-input type="text" class="newsletter__input" placeholder="Email" name="email"
```
Comment (damienwebdev) [inline apps/demo/src/app/newsletter/newsletter.component.html]: You should render an error message here if the email isn't a valid email.

File: apps/demo/src/environments/environment.ts [hunk 11]
```diff
@@ -8,7 +8,7 @@ export const environment = {
   useMocks: true,
   e2e: false,
   name: 'local',
-  driver: 'MAGENTO',
+  driver: 'INMEMORY',
```
Comment (damienwebdev) [inline apps/demo/src/environments/environment.ts]: revert.

**Comment by damienwebdev** [comment]
@Nolan-Arnold this fails linting.

**Pull request merged**